### ISSUE TITLE
Comparison results do not compare correctly to boolean literals or variables

### ISSUE THREAD

**Issue by pcahyna** [pallets/jinja#813]
## Expected Behavior
Results of comparisons should be possible to compare against boolean literals or variables. E.g. status == status should be true for any value of the variable status, so comparing it to true should yield true.

## Actual Behavior
Results of comparisons compare incorrectly to booleans. status == status is true, but comparing it to true yields false. Interestingly, comparing (status == status) to itself gives true, as it should.

## Template Code
```python
from jinja2 import Template
template = Template('{{ status == status }}, {{ (status == status) == constant}}, {{ (status == status) == true}}, {{ (status == status) == (status == status)}}')
template.render(status="a", constant=True)
```
gives
u'True, False, False, True'
## Your Environment
* Python version: 2.7.14
* Jinja version: 2.10

**Comment by asenov** [comment]
This is a fixable issue, but quick workaround is the following code:
 
``from jinja2 import Template
template = Template('{{ status == status }}, {{ (status == status) and constant}}, {{ (status == status) and true}}, {{ (status == status) == (status == status)}}')
r = template.render(status="a", constant=True)
```
It gives: 
True, True, True, True

Regards

**Comment by pcahyna** [comment]
This workaround works only in my artificial example, where the LHS was given as (status == status) and so always true. If we have a more realistic expression as the LHS, whose value can be either true or false, "==" is not equivalent to "and" anymore.

**Comment by ricky-undeadcoders** [comment]
I daresay this is the same issue as #755, and can be closed as a duplicate.

**Comment by davidism** [comment]
Fixed by #938
